checkov/kubernetes/graph_builder/graph_components/edge_builders/KeywordEdgeBuilder.py

```diff
@@ -0,0 +1,56 @@
+from checkov.kubernetes.kubernetes_utils import FILTERED_RESOURCES_FOR_EDGE_BUILDERS
+from checkov.common.util.type_forcers import force_list
+
+
+class KeywordEdgeBuilder(K8SEdgeBuilder):
+
+    @staticmethod
+    def should_search_for_edges(vertex: KubernetesBlock) -> bool:
+        return vertex.attributes.get("kind") in ResourceKeywordIdentifier.KINDS_KEYWORDS_MAP.keys() \
+            and vertex.attributes.get("kind") not in FILTERED_RESOURCES_FOR_EDGE_BUILDERS
+
+    @staticmethod
+    def find_connections(vertex: KubernetesBlock, vertices: list[KubernetesBlock]) -> list[int]:
+        """
+        connection is defined by a match between a vertex's (of a certain type) references definitions to a potential
+        vertex (of any type).
+
+        example:
+        A Pod with the property 'spec.serviceAccountName' with value 'service-123' will match a resource of type
+        'ServiceAccount' with a 'metadata.name' property equals to 'service-123'
+        """
+
+        connections: list[int] = []
+        for potential_vertex_index, potential_vertex in enumerate(vertices):
+            if potential_vertex.id == vertex.id:
+                continue
+            references_definitions: list[dict[str, str]] = ResourceKeywordIdentifier.KINDS_KEYWORDS_MAP[vertex.attributes["kind"]]  # type: ignore[assignment]
+            for references_definition in references_definitions:
+                match = True
+                # check that resource items comply to all references definitions defined in ResourceKeywordIdentifier
+                for reference_key, reference_value in references_definition.items():
+                    reference_is_list = True if isinstance(reference_value, list) else False
+                    reference_values_list = force_list(reference_value)
+                    # some items are nested in lists and their value in the vertex is concatenated with their index,
+                    # like so:  subjects.0.name
+                    # the following lines force all properties to a list for ease of use and concatenate the index
+                    # only to items that are actual lists
+                    for i, reference_value_item in enumerate(reference_values_list):
+                        if reference_is_list:
+                            reference_value_item = f".{i}.".join(reference_value_item.split(".", 1))
+                        vertex_ref = vertex.attributes.get(reference_value_item)
+                        potential_vertex_ref = potential_vertex.attributes.get(reference_key)
+                        if vertex_ref is None or potential_vertex_ref is None or vertex_ref != potential_vertex_ref:
+                            # if not all attributes match then it's not qualified as an edge
+                            match = False
+                            break
+                if match:
+                    connections.append(potential_vertex_index)
+                    break
```

Comment:
If I understand the logic correctly, as long as we don't match all attributes, we want to append to the connection?

Reply:
Only if we match all attributes for a single object in `KINDS_KEYWORDS_MAP[resource]` list - we append a connection